src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -117,24 +119,35 @@ export class AccountManagementService implements IAccountManagementService {
 	 * @return Promise to return an account
 	 */
 	public addAccount(providerId: string): Thenable<void> {
-		let self = this;
+		const loginNotification: INotification = {
+			severity: Severity.Info,
+			message: localize('loggingIn', "Adding account..."),
+			progress: {
+				infinite: true
+			}
+		};
 
 		return this.doWithProvider(providerId, async (provider) => {
-			let account = await provider.provider.prompt();
-			if (self.isCanceledResult(account)) {
-				return;
-			}
+			const notificationHandler = this._notificationService?.notify(loginNotification);
```

Comment:
What circumstances would you expect this to not exist? Just for the test case? There's a TestNotificationService you can inject instead so we don't have to check for that (since we shouldn't be needing to)

https://github.com/microsoft/azuredatastudio/blob/master/src/vs/platform/notification/test/common/testNotificationService.ts#L9

Reply:
Yeah the test case, I can use the testNotificationService